shared/src/main/java/uk/gov/di/authentication/shared/entity/UserProfile.java

```diff
@@ -147,90 +174,64 @@ public UserProfile setClientConsent(ClientConsent consent) {
         return this;
     }
 
-    @DynamoDBIndexHashKey(
-            globalSecondaryIndexName = "PublicSubjectIDIndex",
-            attributeName = ATTRIBUTE_PUBLIC_SUBJECT_ID)
+    @DynamoDbSecondaryPartitionKey(indexNames = {"PublicSubjectIDIndex"})
+    @DynamoDbAttribute(ATTRIBUTE_PUBLIC_SUBJECT_ID)
     public String getPublicSubjectID() {
         return publicSubjectID;
     }
 
-    public UserProfile setPublicSubjectID(String publicSubjectID) {
+    public void setPublicSubjectID(String publicSubjectID) {
+        this.publicSubjectID = publicSubjectID;
+    }
+
+    public UserProfile withPublicSubjectID(String publicSubjectID) {
         this.publicSubjectID = publicSubjectID;
         return this;
     }
 
-    @DynamoDBAttribute(attributeName = ATTRIBUTE_LEGACY_SUBJECT_ID)
+    @DynamoDbAttribute(ATTRIBUTE_LEGACY_SUBJECT_ID)
     public String getLegacySubjectID() {
         return legacySubjectID;
     }
 
-    public UserProfile setLegacySubjectID(String legacySubjectID) {
+    public void setLegacySubjectID(String legacySubjectID) {
+        this.legacySubjectID = legacySubjectID;
+    }
+
+    public UserProfile withLegacySubjectID(String legacySubjectID) {
         this.legacySubjectID = legacySubjectID;
         return this;
     }
 
-    @DynamoDBAttribute(attributeName = ATTRIBUTE_SALT)
+    @DynamoDbAttribute(ATTRIBUTE_SALT)
     public ByteBuffer getSalt() {
         return salt;
     }
 
-    public UserProfile setSalt(ByteBuffer salt) {
+    public void setSalt(ByteBuffer salt) {
         this.salt = salt;
-        return this;
     }
 
-    public UserProfile setSalt(byte[] salt) {
-        this.salt = ByteBuffer.wrap(salt);
+    public void setSalt(byte[] salt) {
+        this.salt = ByteBuffer.wrap(salt).asReadOnlyBuffer();
+    }
+
+    public UserProfile withSalt(ByteBuffer salt) {
+        this.salt = salt;
         return this;
     }
 
-    @DynamoDBAttribute(attributeName = ATTRIBUTE_ACCOUNT_VERIFIED)
-    public Boolean getAccountVerified() {
+    @DynamoDbAttribute(ATTRIBUTE_ACCOUNT_VERIFIED)
+    public int getAccountVerified() {
         return accountVerified;
     }
 
-    public UserProfile setAccountVerified(Boolean accountVerified) {
+    public void setAccountVerified(int accountVerified) {
         this.accountVerified = accountVerified;
-        return this;
     }
 
-    @Override
```

Comment:
See the 'toItem' methods have been removed.  Does v2 improve transaction handling?